tools/ci_build/provider_ops_disabler.py

```diff
@@ -0,0 +1,380 @@
+
+        if domain not in operators or\
+           op_type not in version_map[domain]:
+            log.info("Disable {0} {1} {2} {3}".format(domain, opset_from, opset_to, op_type))
+            return True
+
+        opset_from = int(opset_from)
+        raw_opset_to = opset_to
+        opset_to = int(opset_to)
+
+        first_bigger_at = find_first_bigger(version_map[domain][op_type], opset_to)
+        if first_bigger_at == len(version_map[domain][op_type]):
+            opset_to = 999
+        elif opset_from == opset_to or opset_to not in version_map[domain][op_type]:
+            opset_to = version_map[domain][op_type][first_bigger_at]
+
+        for opset in [str(i) for i in range(opset_from, opset_to+1)]:
+            if opset in operators[domain] and\
+               op_type in operators[domain][opset]:
+                return False
+
+        log.info("Disable {0} {1} {2} {3}".format(domain, opset_from, raw_opset_to, op_type))
+        return True  # end of disable_op(...)
+
+    def process_lines(lines, offset, end_marks, call_back):
+        '''extract op info from a logic code line start from offset to the line end
+           with any of end_marks, then trigger callback(op_type, opset_from, opset_to, domain)
+           return next line offset and whether current lines are disabled
+        '''
+
+        end_mark = ''
+        lines_to_process = []
+
+        while True:  # collect the logical code line
+
+            lines_to_process.append(lines[offset])
+            stripped = lines[offset].strip()
+            line_end = False
+
+            for mark in end_marks:
+                if stripped.endswith(mark):
+                    end_mark = mark
+                    line_end = True
+                    break
+
+            if line_end:
+                break
+
+            offset += 1  # end of while
+
+        code_line = ''.join([line.strip() for line in lines_to_process])
+
+        call_back_ret = False
+        if onnx_op in code_line:
+            # e.g. class ONNX_OPERATOR_KERNEL_CLASS_NAME(
+            #          kCpuExecutionProvider, kOnnxDomain, 1, Transpose);
+            # e.g. BuildKernelCreateInfo<ONNX_OPERATOR_KERNEL_CLASS_NAME(
+            #          kCpuExecutionProvider, kOnnxDomain, 7, Cos)>,
+            trim_at = code_line.index(onnx_op) + onnx_op_len
+            *_, domain, opset, op_type =\
+                [arg.strip() for arg in code_line[trim_at: -len(end_mark)].split(',')]
+            call_back_ret = call_back(op_type, opset, opset, domain)
+
+        elif onnx_typed_op in code_line:
+            # e.g. class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(
+            #          kCpuExecutionProvider, kOnnxDomain, 8, float, Expand);
+            # e.g. BuildKernelCreateInfo<ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(
+            #          kCpuExecutionProvider, kOnnxDomain, 7, double, Sin)>,
+            trim_at = code_line.index(onnx_typed_op) + onnx_typed_op_len
```

Comment:
Would be helpful to include the type info (in this example it would be the 'double') when calling call_back so the log messages can include that. It's confusing for a user to see from the log that a specific op looks to be disabled multiple times as there's one output line per typed registration but no mention of the type in it.

e.g. this is the output for disabling the 3 typed registrations of Crop. Makes it look like the script is incorrect and is doing the same thing 3 times.

2020-08-21 11:58:40,562 Build [INFO] - Disable kOnnxDomain 1 1 Crop
2020-08-21 11:58:40,562 Build [INFO] - Disable kOnnxDomain 1 1 Crop
2020-08-21 11:58:40,563 Build [INFO] - Disable kOnnxDomain 1 1 Crop